packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -309,3 +309,98 @@ describe("SessionTests - Accept a session without passing sessionId and receive
     await testComplete_batching();
   });
 });
+
+describe("SessionTests - getState and setState in Session enabled Queues/Subscriptions", function(): void {
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function testGetSetState(): Promise<void> {
+    const sender = senderClient.getSender();
+    await sender.send(testMessagesWithDifferentSessionIds[0]);
+    await sender.send(testMessagesWithDifferentSessionIds[1]);
+    await delay(4000);
+
+    let receiver = await receiverClient.getSessionReceiver({ sessionId: testSessionId1 });
+    let msgs = await receiver.receiveBatch(3);
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(
+      testMessagesWithDifferentSessionIds.some(
+        (x) => msgs[0].messageId === x.messageId && msgs[0].sessionId === x.sessionId
+      ),
+      true,
+      "Received Message doesnt match any of the test messages"
+    );
+
+    let testState = await receiver.getState();
+    should.equal(testState, "");
+
+    await msgs[0].complete();
+    await receiver.close();
+
+    await testPeekMsgsLength(receiverClient, 1);
+
+    receiver = await receiverClient.getSessionReceiver();
+    msgs = await receiver.receiveBatch(2);
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(
+      testMessagesWithDifferentSessionIds.some(
+        (x) => msgs[0].messageId === x.messageId && msgs[0].sessionId === x.sessionId
+      ),
+      true,
+      "Received Message doesnt match any of the test messages"
+    );
+
+    testState = await receiver.getState();
+    should.equal(testState, "");
+    await receiver.setState("new_state");
+    testState = await receiver.getState();
+    should.equal(testState, "new_state");
+
+    await msgs[0].complete();
+    await receiver.setState(""); // clearing the session-state
+    await testPeekMsgsLength(receiverClient, 0);
```

Comment:
What we are doing here is:
- send msg to different sessions
- test that one of them has empty state
- test second one has empty state
- set state on second one
- check state on second one.

Do we really need to work with 2 sessions here?
How about the below:

- Send 1 messages to a session
- create session receiver 
- check that state is empty
- set state and then check state
- close the receiver

Reply:
Also, do we have to verify that when we re-open the session, the state remains same ?

Reply:
That's a good case to test